### DISCUSSION TITLE
How to set the precision of useCurrent and useCurrentOnUpdate

### DISCUSSION THREAD

**Discussion by jyxjjj** [laravel/framework#46640] [Q&A] [answered]
How can we set a column as "ON UPDATE CURRENT_TIMESTAMP(3)";

**Comment by henzeb** [comment]
I don't think you need to. Have you tried without and tested what happens?

**Reply by jyxjjj** [reply]
I am using the latest MariaDB, I tested only using the SQL instead of Migrations, and set timestamps to null in models, it can successfully give the milliseconds even the microseconds, and even a better performance than models' timestamps.
I just want to use the migrations instead of the native SQLs.

**Comment by Angel5a** [comment, marked as answer]
Hi, @jyxjjj

You can specify $precision argument when you define a column. The exactly same precision is used in CURRENT_TIMESTAMP().
`$table->timestamp('precise_timestamp_column', 3)->useCurrentOnUpdate();`

**Reply by jyxjjj** [reply]
Then it will all be the ".000".
The column is a timestamp column with 3 precision, but the "on update" is still the CURRENT_TIMESTAMP instead of CURRENT_TIMESTAMP(3)

**Reply by Angel5a** [reply]
Just took a clear installation.
```
$table->timestamp('precise_timestamp_column', 3)->useCurrent()->useCurrentOnUpdate();
```
Got as expected:
```
> DB::insert('insert into `precision`(string) values ("one")');
= true

> DB::insert('insert into `precision`(string) values ("two")');
= true

> DB::select('select * from `precision`');
= [
    {#3745
      +"id": 1,
      +"string": "one",
      +"precise_timestamp_column": "2023-04-04 19:20:55.940",
      +"created_at": null,
      +"updated_at": null,
    },
    {#3736
      +"id": 2,
      +"string": "two",
      +"precise_timestamp_column": "2023-04-04 19:21:06.580",
      +"created_at": null,
      +"updated_at": null,
    },
  ]

> DB::update('update `precision` set string="three" where string="two"');
= 1

> DB::select('select * from `precision`');
= [
    {#3755
      +"id": 1,
      +"string": "one",
      +"precise_timestamp_column": "2023-04-04 19:20:55.940",
      +"created_at": null,
      +"updated_at": null,
    },
    {#3733
      +"id": 2,
      +"string": "three",
      +"precise_timestamp_column": "2023-04-04 19:22:11.163",
      +"created_at": null,
      +"updated_at": null,
    },
  ]
```
Am I wrong and you are looking for something else?

**Reply by jyxjjj** [reply]
When i ask this question, i had tested this for months ago, that you said is wrong but i forget why.
When you answered, today i tested again, and yes, it is correct.
Thank you. Maybe i should test every time before ask questions.

**Reply by Angel5a** [reply]
@jyxjjj, It's ok, don't worry. Have a nice day.

**Comment by jyxjjj** [comment]
Even I think it is an correct answer, but i don't like laravel's migrations at all any more.
I continue to use .sql file to do database init operations.
So I will mark this as outdated instead of resolved.
